### DISCUSSION TITLE
Flickr pictures not showing any more

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#634] [Q&A]
Hello,

The Flickr pictures aren't showing any more on my Birdnet-Pi installation. The Flickr ID is correctly set up in the settings and I didn't change it anyway. Everything else seems to be fine on my Pi.

What could be the reason?

Thank you.

**Comment by jflyons** [comment]
Hi blogdemoi - I had not installed the Flickr capability on my setup, which I have had going now for a couple of months. Your message prompted me to get it going, and it seems to be working fine. Not much help I know, but glad I have the photos working here. :)

**Reply by blogdemoi** [reply]
> Not much help I know, but glad I have the photos working here. :)

Thank you anyway. :-) Happy to know it is working fine on your Birdnet-Pi. I suspect unplugging the Pi several times didn't do any good to it... I am thinking of buying a case for my Pi with a power off/reboot button. I saw one or two on the net and think it can be useful instead of unplugging the Pi when I can't ssh into. (My Pi disconnects regularly from my router, I followed @mcguirepr89 advice in another thread but it didn't solve my connection problems, I also installed a cron which launches a script to reconnect in case of a WiFi disconnection, but none seem to work unfortunately...).

**Comment by ehpersonal38** [comment]
Change `error_reporting(E_ERROR);` to `error_reporting(E_ALL);` in overview.php and todays_detections.php, then try again on an incognito browser window. You should see a few warning/info/error messages appear at the top of the page

**Reply by blogdemoi** [reply]
So I did the changes and I can see error messages:

> Warning: ini_set(): A session is active. You cannot change the session module's ini settings at this time in /home/birder/BirdNET-Pi/scripts/overview.php on line 4
> 
> Warning: session_set_cookie_params(): Cannot change session cookie parameters when session is active in /home/birder/BirdNET-Pi/scripts/overview.php on line 5
> 
> Notice: session_start(): A session had already been started - ignoring in /home/birder/BirdNET-Pi/scripts/overview.php on line 6
> 
> No Detections For Today
> 
> Notice: Undefined index: only_name in /home/birder/BirdNET-Pi/scripts/overview.php on line 58

**Reply by ehpersonal38** [reply]
Could you take a screenshot of the whole overview page?

**Reply by blogdemoi** [reply]
Two screenshots:

-First one, on the page for a specific bird

![ut0B8bf](https://user-images.githubusercontent.com/83965802/200095149-029cfd17-7a06-483d-928c-c795ad5af8d2.png)

-The whole overview page:

![SMgLskJ](https://user-images.githubusercontent.com/83965802/200095217-9649d91f-3ae2-415c-b8b4-c1a889c0c8c0.png)

**Reply by ehpersonal38** [reply]
Thank you. Could you try these steps?

1) Switch to this debug branch `bash ~/BirdNET-Pi/scripts/update_birdnet.sh -b flickr-debug`
After that, please open the overview page again, wait about 10 seconds for everything to load, and take another screenshot. 

2) Let's also make sure your API key is receiving the correct response from Flickr, try accessing this URL in your browser, but replace the XXXXXXXXXXXXXXXXXXXXXXX with your Flickr API key.
https://www.flickr.com/services/rest/?method=flickr.photos.search&api_key=XXXXXXXXXXXXXXXXXXXXXXX&text=northern%20cardinal&per_page=5&media=photos&format=json&nojsoncallback=1

You should get a response like this if all is working correctly:
```
{"photos":{"page":1,"pages":23553,"perpage":5,"total":117765,"photo":[{"id":"52281929068","owner":"183506946@N03","secret":"d89cb0e55c","server":"65535","farm":66,"title":"Northern Cardinal","ispublic":1,"isfriend":0,"isfamily":0},{"id":"52281864331","owner":"96366044@N02","secret":"bf9fbffc79","server":"65535","farm":66,"title":"Northern Cardinal","ispublic":1,"isfriend":0,"isfamily":0},{"id":"52281768731","owner":"49563725@N06","secret":"02fe85270d","server":"65535","farm":66,"title":"Disagreement","ispublic":1,"isfriend":0,"isfamily":0},
```

Hopefully that'll give us more information and we can proceed from there.

**Reply by blogdemoi** [reply]
Hello,

I have this (screenshot below):

> Warning: ini_set(): A session is active. You cannot change the session module's ini settings at this time in /home/birder/BirdNET-Pi/scripts/overview.php on line 4
> 
> Warning: session_set_cookie_params(): Cannot change session cookie parameters when session is active in /home/birder/BirdNET-Pi/scripts/overview.php on line 5
> 
> Notice: session_start(): A session had already been started - ignoring in /home/birder/BirdNET-Pi/scripts/overview.php on line 6
> 
> No Detections For Today
> array(2) { ["behind"]=> string(4) "with" ["images"]=> array(1) { [0]=> array(5) { [0]=> string(15) "Hibou_moyen-duc" [1]=> string(37) "https://farm.static.flickr.com//_.jpg" [2]=> NULL [3]=> string(27) "https://flickr.com/photos//" [4]=> string(26) "https://flickr.com/people/" } } } 1: SESSION is NOT null
> array(2) { ["behind"]=> string(4) "with" ["images"]=> array(1) { [0]=> array(5) { [0]=> string(15) "Hibou_moyen-duc" [1]=> string(37) "https://farm.static.flickr.com//_.jpg" [2]=> NULL [3]=> string(27) "https://flickr.com/photos//" [4]=> string(26) "https://flickr.com/people/" } } }
> 
> Notice: Undefined index: only_name in /home/birder/BirdNET-Pi/scripts/overview.php on line 62
> 2: 0

![awJNMk1](https://user-images.githubusercontent.com/83965802/200207144-81c867be-baf2-4062-96e3-a4e1a166fcb7.png)

As for my Flickr API key, I followed your instructions and I have this:

![xyCv3G5](https://user-images.githubusercontent.com/83965802/200207393-14b2baec-5b0d-4341-b4ed-029dca7aeb29.png)

Thank you for your help.

**Reply by ehpersonal38** [reply]
Thank you! 

I added some more debug info to the test branch, so please update: `bash ~/BirdNET-Pi/scripts/update_birdnet.sh -b flickr-debug`

After updating if you're able to grab a screenshot of the home page of your BirdNET-Pi again, but this time on a incognito/private window, or clear all your cache and cookies on your regular browser. 

I'm interested in the output of the very first load, so please don't refresh the page or navigate away and then back again.

**Reply by blogdemoi** [reply]
From a Firefox private window, after updating:

![OLo3rXM](https://user-images.githubusercontent.com/83965802/200211959-42abcade-034b-4611-9c35-7bed74d74233.png)

> birdnetpi:~/BirdNET-Pi$ bash ~/BirdNET-Pi/scripts/update_birdnet.sh -b flickr-debug
> + sudo -u birder git -C /home/birder/BirdNET-Pi reset --hard
> HEAD is now at fe7426a Update overview.php
> + set +x
> + sudo -u birder git -C /home/birder/BirdNET-Pi fetch origin flickr-debug
> fatal: unable to access 'https://github.com/mcguirepr89/BirdNET-Pi.git/': Could not resolve host: github.com
> + set +x
> + sudo -u birder git -C /home/birder/BirdNET-Pi switch -C flickr-debug --track origin/flickr-debug
> Branch 'flickr-debug' set up to track remote branch 'flickr-debug' from 'origin'.
> Reset branch 'flickr-debug'
> Your branch is up to date with 'origin/flickr-debug'.
> + set +x
> + my_dir=/home/birder/BirdNET-Pi/scripts
> + sudo systemctl stop birdnet_server.service
> + sudo systemctl stop birdnet_recording.service
> + services=(chart_viewer.service spectrogram_viewer.service icecast2.service extraction.service birdnet_recording.service birdnet_log.service)
> + for i in "${services[@]}"
> + sudo systemctl restart chart_viewer.service
> + for i in "${services[@]}"
> + sudo systemctl restart spectrogram_viewer.service
> + for i in "${services[@]}"
> + sudo systemctl restart icecast2.service
> + for i in "${services[@]}"
> + sudo systemctl restart extraction.service
> + for i in "${services[@]}"
> + sudo systemctl restart birdnet_recording.service
> + for i in "${services[@]}"
> + sudo systemctl restart birdnet_log.service
> + sudo systemctl start birdnet_server.service
> + sleep 5
> + for i in {1..5}
> + systemctl is-active --quiet birdnet_server.service
> + grep 5050 /dev/fd/63
> ++ netstat -tulpn
> + sleep 5
> + for i in {1..5}
> + systemctl is-active --quiet birdnet_server.service
> + grep 5050 /dev/fd/63
> ++ netstat -tulpn
> + sleep 5
> + for i in {1..5}
> + systemctl is-active --quiet birdnet_server.service
> + grep 5050 /dev/fd/63
> ++ netstat -tulpn
> + sleep 5
> + for i in {1..5}
> + systemctl is-active --quiet birdnet_server.service
> + grep 5050 /dev/fd/63
> ++ netstat -tulpn
> + sleep 5
> + for i in {1..5}
> + systemctl is-active --quiet birdnet_server.service
> + grep 5050 /dev/fd/63
> ++ netstat -tulpn
> + sleep 5
> + systemctl is-active --quiet birdnet_server.service
> + grep 5050 /dev/fd/63
> ++ netstat -tulpn
> + status=1
> + ((  status != 0  ))
> + logger '[/usr/local/bin/restart_services.sh] Unable to start birdnet_server.service... Looping until it start properly'
> + grep 5050 /dev/fd/63
> ++ netstat -tulpn
> + sudo systemctl restart birdnet_server.service
> + sleep 45
> + grep 5050 /dev/fd/63
> ++ netstat -tulpn
> tcp        0      0 127.0.0.1:5050          0.0.0.0:*               LISTEN      29193/python3
> + sudo systemctl restart birdnet_analysis.service
> birdnetpi:~/BirdNET-Pi$

**Reply by ehpersonal38** [reply]
After seeing the message `Could not resolve host: github.com` I'm pretty sure this is an issue with your system not resolving DNS hostnames properly. Does running `host flickr.com` produce an IP address? If not then I would lookup how to configure your DNS on a Raspberry Pi. Could be a proxy getting in the way too, if you use one.

**Reply by blogdemoi** [reply]
You were right. I should have checked this, I am very sorry for having taken your time. I checked  /etc/resolv.conf and changed the value for nameserver for another one and the Flickr images are back. 

I think /etc/resolv.conf  is the right place to change DNS... Am I right?

Now that it is fixed how can I disable the error messages on the overview page? Just reverting "Change error_reporting(E_ERROR); to error_reporting(E_ALL); in overview.php and todays_detections.php"?

Thank you.

**Reply by ehpersonal38** [reply]
No worries, I learned something too. I think resolv.conf might get reset each reboot (not sure though), I'd Google for the proper way to set DNS on a raspberry Pi. 

You can revert by updating your system to the main branch: `bash ~/BirdNET-Pi/scripts/update_birdnet.sh`

Take care.

**Comment by Papasean** [comment]
Sorry if I'm asking a question that has already been asked......but as this project is so heavily tied to the "All About Birds" platform why are the images not sourced from there?

**Reply by ehpersonal38** [reply]
As far as I'm aware, BirdNET-Pi doesn't have any ties to any of the Cornell platforms or projects. But I suppose we could've used that platform (which uses Macaulay Library images), just never occured to me.
